docs/concepts/juttle_tutorial.md

```diff
@@ -2,27 +2,532 @@
+
+```raw
+┌──────────────────────────────────────────────────────┐
+│ fields                                               │
+├──────────────────────────────────────────────────────┤
+│ type,actor_login,repo_name,n_commits                 │
+└──────────────────────────────────────────────────────┘
+```
+
+To see the set of unique values for a given field, use `reduce by` field name, shown here for field `type`. As you may not know in advance how many unique values there will be, it is advisable to first run the program with `reduce count()` tacked on, and once you know that the output is reasonably small, view the values themselves.
+
+```
+read file -file '/tmp/data.json' -from :2015-07-01: -to :2015-07-02:
+| reduce by type
+| reduce count()
+```
+
+As there are mere 10 types, it's fine to output them in a table:
+
+```
+read file -file '/tmp/data.json' -from :2015-07-01: -to :2015-07-02:
+| reduce by type
+```
+
+Since we didn't use `pluck` to make an array, the output will be multiple data points, one for each type (only showing a couple here):
+
+```
+┌─────────────────┐
+│ type            │
+├─────────────────┤
+│ ForkEvent       │
+├─────────────────┤
+...
+├─────────────────┤
+│ CreateEvent     │
+└─────────────────┘
+```
+
+Now we have a good idea of what individual data points look like in terms of fields and their values, but don't yet grok the size and shape of the data: how many data points there are, what is the distribution of values in a given time interval, and over time. This is where Juttle shines.
+
+## How many repos? (Different reducers)
+
+Recall that there is a field called "repo_name" in the data. It might be useful to know how many unique repository names we have. To do this, we'll
+use the [`count_unique(..)`](../reducers/count_unique.md) reducer.
+
+```
+read file -file '/tmp/data.json'
+| reduce repos = count_unique('repo_name')
+```
+
+This reducer takes a parameter, which is a _field name_ specifying which field of the incoming points we want to compute the cardinality of.
+
+We can also use `reduce` with multiple reduce expressions. For example, we can
+count the total number of points, the number of unique repository names, and the
+number of unique actors (in the "actor_login" field) like this:
+
+```
+read file -file '/tmp/data.json'
+| reduce total = count(), 
+    actors = count_unique('actor_login'), 
+    repos = count_unique('repo_name')
+```
+
+## Enriching the data (Put vs reduce)
+
+Let's say we wanted to know the number of actors participating in an average GitHub repository. The program above gives us the counts of actors and repos, so finding the average is simple math. However, it will not work to add "avg_actors = actors / repos" to the `reduce` expression, since this computation is not a reducer; and in general, this is not the logic you seek.
```

Comment:
I had trouble with this bit, there's a conversation going on gitter, https://gitter.im/juttle/juttle?at=5682d4180199d70069e01f47
My best idea right now is to simply link from this to the definition of our reducers.